### DISCUSSION TITLE
Filter by relation id

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#1539] [Q&A] [answered]
I have a collection of songs that have a relation to a user, I now want to fetch all songs from a user by the user id.

I try to do it like this, from my understanding it should take the current logged in user id and filter the songs collection with it:

`      const resultList = await this.pb.collection('songs').getList(1, 50, {
        filter: `user.id=${this.pb.authStore.model?.id}`,`

However I receive this error:
`GEThttp://127.0.0.1:8090/api/collections/songs/records?page=1&perPage=50&filter=user.id=den82a2xvkq7ng4`

> ClientResponseError 400: Invalid filter parameters.

Filtering in the backend itself seems to work:
![image](https://user-images.githubusercontent.com/10931525/211214772-652bff0d-6c6e-49bf-a413-d378494058df.png)

These are my API rules, CUD should be only possible when the related user is in the auth header, reading should be public
![image](https://user-images.githubusercontent.com/10931525/211214776-1a8c141c-1505-4940-8d3a-1b5dafc1184a.png)

What am I missing here?

**Comment by ganigeorgiev** [comment, marked as answer]
The js variable need to be passed as string literal, aka. wrapped with single or double quotes:

```js
const resultList = await this.pb.collection('songs').getList(1, 50, {
    filter: `user.id="${this.pb.authStore.model?.id}"`
    // for single relations you can also just use the field name without specifying the id
    // filter: `user="${this.pb.authStore.model?.id}"`
})
```

**Reply by ghost** [reply]
Ok, well I was close :D Thank you, it's insane how fast you respond.
